src/backend/tests/partaj/core/test_api_referral.py

```diff
@@ -2215,6 +2215,36 @@ def test_unassign_referral_from_closed_state(self, _):
         self.assertEqual(referral.assignees.count(), 1)
 
     # ASSIGN UNIT TESTS
+
+    def test_assign_unit_with_missing_explanation(self, mock_mailer_send):
+        """
+        Assign unit explanation is mandatory. Make sure the API returns an error when
+        it is missing.
+        """
+        referral = factories.ReferralFactory(state=models.ReferralState.RECEIVED)
+        user = factories.UnitMembershipFactory(
+            role=models.UnitMembershipRole.OWNER, unit=referral.units.get()
+        ).user
+        other_unit = factories.UnitFactory()
+
+        response = self.client.post(
+            f"/api/referrals/{referral.id}/assign_unit/",
+            {"unit": str(other_unit.id)},
+            HTTP_AUTHORIZATION=f"Token {Token.objects.get_or_create(user=user)[0]}",
+        )
+
+        self.assertEqual(response.status_code, 400)
+        referral.refresh_from_db()
+        self.assertEqual(
```

Comment:
Also make sure there is still only one unit assigned to this referral (referral.units.count() == 1)